compiler/rustc_builtin_macros/src/cfg_eval.rs

```diff
@@ -166,7 +166,9 @@ impl CfgEval<'_, '_> {
                     ))
                 },
                 Annotatable::Stmt(_) => |parser| {
-                    Ok(Annotatable::Stmt(P(parser.parse_stmt(ForceCollect::Yes)?.unwrap())))
+                    Ok(Annotatable::Stmt(P(parser
+                        .parse_stmt_without_recovery(false, ForceCollect::Yes)?
```

Comment:
Not that familiar with the macro expansion code, so feel free to re-assign. 

But I'm wondering why we even get into the `expand` call if the parse for the `Annotatable` failed in the first place. If I understand this correctly this parse is a re-parse (we also get a duplicated error message in the test). Why can't we prevent this to be called when the first parse failed?